src/java.base/share/classes/jdk/internal/classfile/impl/AbstractPoolEntry.java

```diff
@@ -380,6 +382,7 @@ else if ((state == State.STRING && u.state == State.STRING))
 
         @Override
         public boolean equalsString(String s) {
+            Objects.requireNonNull(s);
```

Comment:
I’d prefer if this method returned `false` on `null`, just like how `Object​::equals​(Object)` and `String​::equalsIgnoreCase​(String)` return `false` on `null`.

Reply:
We should have noted clearly that the intended use case is for a read utf8 to compare to an existing string, like a method name, so we can minimally inflate a utf8 to perform a comparison. This means that there is little purpose of using `null` here, just like for any other Class-File API endpoints.

Reply:
Noted, if this is the expected output by users.
I will mention this as a talking point when opening the PR.